core/dbt/cli/params.py

```diff
@@ -40,6 +40,14 @@
     default=True,
 )
 
+compile_suppress_ephemeral_ctes = click.option(
+    "--suppress-ephemeral-ctes/--no-suppress-ephemeral-ctes",
```

Comment:
From review with @graciegoheen, we discussed naming this option more consistently with existing naming conventions, where the default behaviour (in this case - injecting ephemeral ctes) is effectively the flag name, which also has a negated version. For example `--partial-parse/--no-partial-parse`. 

In this case, we could consider: 
`--inject-ephemeral-ctes/--no-inject-ephemeral-ctes`

@benmosher - What do you think?

Reply:
I'd like to go with `--inject-ephemeral-ctes/--no-inject-ephemeral-ctes`, so that it's clear what the default is `--inject-ephemeral-ctes` vs. the override `--no-inject-ephemeral-ctes`. 

I worry with something like `--lint-mode`, folks may get confused that dbt is doing the actual linting.